<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="type" default="keyboard"/>

    <group if="$(eval type == 'gamepad')">
        <node respawn="true" pkg="joy" type="joy_node" name="gamped_ros_driver">
            <param name="dev" type="string" value="/dev/input/js0" />
            <param name="deadzone" value="0.12" />
        </node>

        <rosparam command="load" file="$(find go_car_manual_control)/config/gamepad_driver_params.yaml"/>
        <node pkg="go_car_manual_control" type="gamepad_controller_adapter" name="gamepad_controller_adapter"/>
    </group>

    <group if="$(eval type == 'keyboard')">
        <node pkg="go_car_manual_control" type="keyboard_listener" name="keyboard_listener"/>
        <node pkg="go_car_manual_control" type="keyboard_input_processor" name="keyboard_input_processor"/>
    </group>

</launch>
